src/Courses/CourseProvider.tsx

```diff
@@ -0,0 +1,71 @@
+import { Backdrop, CircularProgress } from '@material-ui/core';
+import React, { useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+import { getUsersForCourse } from '../APIInterfaces/BackendAPI/Requests/UserRequests';
+import AxiosRequest from '../Hooks/AxiosRequest';
+import { CourseObject, UserObject } from './CourseInterfaces';
+
+interface CourseProviderProps {
+    children: React.ReactNode
+}
+
+const CourseContext = React.createContext<{
+        course: CourseObject, 
+        setCourse: React.Dispatch<React.SetStateAction<CourseObject>> | undefined,
+        error: string | null,
+        users: UserObject[],
+        setUsers: React.Dispatch<React.SetStateAction<UserObject[]>> | undefined,
+    }>({
+        course: new CourseObject(), 
+        setCourse: undefined,
+        error: null,
+        users: [],
+        setUsers: undefined,
+    });
+
+export const useCourseContext = () => React.useContext(CourseContext);
+
+export const CourseProvider: React.FC<CourseProviderProps> = ({children}) => {
+    const { courseId } = useParams<{courseId: string | undefined}>();
```

Comment:
In this case you can't use the question mark

Reply:
Actually, I think you can use the question mark because it's in an object.